Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -12,14 +12,18 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
             var moves = new List<Square>();
             var currentSquare = board.FindPiece(this);
-            moves.Add(this.Player == Player.White
-                ? Square.At(currentSquare.Row - 1, currentSquare.Col)
-                : Square.At(currentSquare.Row + 1, currentSquare.Col));
-            if (!HasMoved)
+            var playerDirection = this.Player == Player.White ? -1 : 1;
+            for (int i = 1; i < 3; i++)
```

Comment:
So you're just doing this loop twice? In that case I would just write it out, as using a loop just makes it harder to understand 🙂